src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,173 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    static class Node<T> {
+        private Node<T> prev;
+        private T value;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (first == null) {
+            first = last = newNode;
+        } else {
+            last.next = newNode;
+            newNode.prev = last;
+            last = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("U cant get element by this " + index);
+        }
+        Node<T> newNode = new Node<>(null, value, null);
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
+            newNode.next = first;
+            first.prev = newNode;
+            first = newNode;
+            size++;
+        } else {
+            Node<T> nodeByIndex = findNodeByIndex(index);
+            newNode = new Node<>(nodeByIndex.prev, value, nodeByIndex);
+            nodeByIndex.prev.next = newNode;
+            nodeByIndex.prev = newNode;
+            size++;
+        }
+    }
+
+    Node<T> findNodeByIndex(int index) {
+        if (index < (size >> 1)) {
+            Node<T> head = first;
+            for (int i = 0; i < index; i++) {
+                head = head.next;
+            }
+            return head;
+        } else {
+            Node<T> tail = last;
+            for (int i = size - 1; i > index; i--) {
+                tail = tail.prev;
+            }
+            return tail;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T lst : list) {
+            add(lst);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        exception(index);
+        if (index == 0) {
+            return first.value;
+        }
+        return findNodeByIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        exception(index);
+        if (index == 0) {
+            T headValue = first.value;
+            Node<T> newNode = new Node<>(null, value, first.next);
```

Comment:
we can directly replace value in existing node 
we shouldn't create new Node and change the links